unix/main.c

```diff
@@ -338,10 +338,17 @@ STATIC void pre_process_options(int argc, char **argv) {
                         heap_size *= 1024;
                     } else if ((*end | 0x20) == 'm') {
                         heap_size *= 1024 * 1024;
+                    } else if (*end != '\0') {
+                        mp_printf(&mp_stderr_print, "Unrecognized heapsize modifier: %c\n", *end);
+                        exit(usage(argv));
                     }
                     if (word_adjust) {
                         heap_size = heap_size * BYTES_PER_WORD / 4;
                     }
+                    if (heap_size < 1024) {
```

Comment:
uPy perfectly well starts up with 0.5K, and generally, lowest size should not be arbitrarily limited.